sycl/include/sycl/ext/intel/experimental/esimd/math.hpp

```diff
@@ -112,12 +122,12 @@ __ESIMD_API std::enable_if_t<std::is_integral<T0>::value &&
                                  std::is_integral<U>::value,
                              __ESIMD_NS::simd<T0, SZ>>
 shr(__ESIMD_NS::simd<T1, SZ> src0, U src1, Sat sat = {}) {
-  using ComputationTy = __ESIMD_DNS::computation_type_t<decltype(src0), U>;
-  typename __ESIMD_DNS::simd_type<ComputationTy>::type Src0 = src0;
-  typename __ESIMD_DNS::simd_type<ComputationTy>::type Src1 = src1;
+  using ComputationTy =
+      __ESIMD_DNS::computation_type_t<decltype(src0), int32_t>;
```

Comment:
For SHR it is important to distinguish signed int32 (arithmetic shift right) vs unsigned int32 (logical shift right).
(unsigned char)(0x80) >> (int)(1) should result into 0x00000040, not 0xFFFFFFC0
Signed-ness should be taken from the 'src0' operand.

Can you please also add such a test case to the LIT test?